packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -1235,21 +1236,27 @@ export class IndexedDbLruDelegate implements ReferenceDelegate, LruDelegate {
     txn: PersistenceTransaction,
     upperBound: ListenSequenceNumber
   ): PersistencePromise<number> {
```

Comment:
Is there a reason we can't create / use a RemoteDocumentChangeBuffer here so that it does the size-tracking instead of us?  That would clean up this code and allow us to potentially completely encapsulate size-tracking to an implementation detail of RemoteDocumentCache/RemoteDocumentChangeBuffer (by making RemoteDocumentChangeBuffer.removeEntry / addEntry private).